packages/medusa/src/api/routes/admin/discounts/delete-condition.ts

```diff
@@ -78,39 +73,34 @@ import { validator } from "../../../../utils/validator"
 export default async (req, res) => {
   const { discount_id, condition_id } = req.params
 
-  const validatedParams = await validator(
-    AdminDeleteDiscountsDiscountConditionsConditionParams,
-    req.query
-  )
-
   const conditionService: DiscountConditionService = req.scope.resolve(
     "discountConditionService"
   )
+  const discountService: DiscountService = req.scope.resolve("discountService")
 
   const condition = await conditionService
     .retrieve(condition_id)
     .catch(() => void 0)
 
   if (!condition) {
+    const discount = await discountService.retrieve(
```

Comment:
thought: This would lead to a non-idempotent delete in case the discount id provided does not exist. I think we should also resolve idempotently for that case. Let me know what you think?

Reply:
I am fine with being idempotent on both entities, but I think that if the discount does not exists we should not be idempotent for that step but only for the condition. It seams a bit weird that both discount and condition does not exists at that point no?

Reply:
We can't control what the consumers provide as ids to the endpoints, so we could have someone do: `DELETE /admin/discounts/bogus_id/conditions/bogus_id_2` and for that case, I would find it reasonable to still resolve idempotently, I am not sure what you think about that?

Reply:
I am fine with that 👍 I was just thinking that from a user point of view it seams that it worked and therefore I will continue, but I am not aware that I have messed up my ids unintentionally by for example adding an extra char because of a bad impl. if you see what I mean

Reply:
@olivermrbl just had a look and the original behaviour was to fail if the discount does not exists. I ll change it, but it changes the API a bit for the end user. Should it be mentioned somewhere?

Reply:
I don't have a strong opinion, so I am also cool with keeping the original behaviour, if you find it better

Reply:
I prefer it indeed, but I did the modification for the new behaviour 😂

Reply:
Feel free to revert 😅